src/models/ElasticDocument.php

```diff
@@ -89,11 +89,14 @@ protected function loadByEntry( craft\elements\Entry $entry )
 
     protected function loadByGlobalSet( craft\elements\GlobalSet $globalSet )
     {
-        $this->type = 'global';
+        // We'll prefix the type in order to avoid conflict with channels/structures
+        $this->type = 'global:' . $globalSet->handle;
```

Comment:
Good point. I've checked and verified that `-` is not allowed as part of the handle in Craft, so using is to separate the prefix from the handle is safe.